lib/commentOnOpenedPullRequest.js

```diff
@@ -0,0 +1,129 @@
+    return commentOnOpenedPullRequest._implementation(pullRequest.url + '/files', pullRequest.comments_url, headers, changesSignature, thirdPartySignature, thirdPartyFolders);
+}
+
+/** Implementation
+ *
+ * @param {String} pullRequestFilesUrl API Url to GET which files changed in pull request (https://developer.github.com/v3/pulls/#list-pull-requests-files)
+ * @param {String} pullRequestCommentsUrl API Url to POST comments to
+ * @param {Object} headers Headers
+ * @param {String} changesSignature Message to post if CHANGES.md has changed in PR
+ * @param {String} thirdPartySignature Message to post if any third party files have changed
+ * @param {String[]} thirdPartyFolders Folder paths to search for third party file changes
+ * @private
+ * @return {Promise<http.IncomingMessage | String>} Response of POST
+ */
+commentOnOpenedPullRequest._implementation = function (pullRequestFilesUrl, pullRequestCommentsUrl, headers,
+                                                       changesSignature, thirdPartySignature, thirdPartyFolders) {
+    return requestPromise.get({
+        url: pullRequestFilesUrl,
+        headers: headers,
+        json: true,
+        resolveWithFullResponse: true
+    })
+    .then(function (filesJsonResponse) {
+        var message = '';
+        var files;
+        var didUpdateChanges;
+        var didUpdateThirdParty;
+
+        if (!statusCodeIsOk(filesJsonResponse)) {
+            return Promise.reject('Bad status code with url ' + pullRequestFilesUrl);
+        }
+        files = filesJsonResponse.body.map(function (file) {
+            return file.filename;
+        });
+        dateLog('These files changed in the pull request: ' + files);
+
+        didUpdateChanges = commentOnOpenedPullRequest._didUpdateChanges(files);
+        didUpdateThirdParty = commentOnOpenedPullRequest._didUpdateThirdParty(files, thirdPartyFolders);
+        if (didUpdateChanges) {
+            dateLog('CHANGES.md was updated');
+            message += changesSignature;
+        }
+        if (didUpdateThirdParty) {
+            dateLog('A third party file changed');
+            message += '\nAlso: ' + thirdPartySignature;
+        }
+        if (didUpdateChanges || didUpdateThirdParty) {
+            return requestPromise.post({
+                uri: pullRequestCommentsUrl,
+                headers: headers,
+                body: {
+                    body: message
+                },
+                json: true,
+                resolveWithFullResponse: true
+            });
+        }
+        return Promise.resolve('Did not change CHANGES.md or any third party files');
+    });
+};
+
+/** Check if CHANGES.md updated
+ *
+ * @param {String[]} files Paths to changed files
+ * @private
+ * @return {boolean} True if any file in `thirdPartyFolders` has changed
+ */
+commentOnOpenedPullRequest._didUpdateChanges = function (files) {
+    for (var i = 0; i < files.length; i++) {
+        if (/^(\.\/)?CHANGES\.md/.test(files[i])) {
```

Comment:
Is there a reason the `./` may only be there some of the time?  It seems weird that we would have to check for both.